### ISSUE TITLE
Modesetting stopped working?

### ISSUE THREAD

**Issue by trombonehero** [FreeBSDDesktop/freebsd-base-graphics#126]
With recent `drm-next` (92f1aeaa3edba87c8665173b99eedfc138c8c734 merged with the EFI loader fixes from upstream) and `xorg-server-1.18.4,1` on Skylake, I see the following error(s) on startx with modesetting:

```
MESA-LOADER: failed to retrieve device information
Require OpenGL version 2.1 or later.
(EE)
Fatal server error:
(EE) AddScreen/ScreenInit failed for driver 0
```

This used to work, but unfortunately I've had to update both my packages and my kernel at the same time recently, so I'm not sure which bit is the chicken and which bit is the egg. Is there any more information I can provide? If I install the `xf86-video-intel` package, the desktop runs but very sluggishly. Running `glxgears` causes the same Mesa error followed by general unresponsiveness (though the power button does, eventually, cause the machine to shut down).

**Comment by jbeich** [comment]
Can you try the patch in [bug 217585](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=217585)? I think it's a regression from freebsd/freebsd-ports@c2a7c09121f5: libdevq bits moved from libGL/libEGL to libdrm.

**Comment by nomadlogic** [comment]
I have been seeing this as well since upstream performed the Xorg upgrade.  My workaround has been to either use modesetting with Option "AccelMethod" "none" to completely disable glamor, or to use the intel driver with either SNA or UXA accel.

I pinged the ports maintainers but haven't gotten any traction.  My plan was to wait until GL libraries are updated upstream and re-test.  If still seeing issue I'd pester more forcefully :)

FWIW - I did have modesetting+glamor working fine with CTF patches when compiled locally, but once I used the official Xorg binaries it stopped working.
